test/parallel/test-child-process-flush-stdio.js

```diff
@@ -20,10 +21,42 @@ function spawnWithReadable() {
     assert.strictEqual(code, 0);
     assert.strictEqual(signal, null);
     assert.strictEqual(Buffer.concat(buffer).toString().trim(), '123');
+    spawnWithPipe();
   }));
   p.stdout.on('readable', function() {
-    let buf;
-    while (buf = this.read())
-      buffer.push(buf);
+    setImmediate(() => {
+      let buf;
+      while (buf = this.read())
+        buffer.push(buf);
+    });
```

Comment:
@petrosagg Can confirm that this test case worked before 12274a5 – nice catch! A separate PR seems like a good idea, yep